bin/json-mask.js

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env node
+const mask = require('../')
+
+function usage (error) {
+  error && console.error(error.message)
+  console.log('Usage: json-mask <fields> [input.json]')
+  console.log('Examples:')
+  console.log('  json-mask "url,object(content,attachments/url)" input.json')
+  console.log('  cat input.json | json-mask "url,object(content,attachments/url)"')
+  console.log('  curl https://api.myjson.com/bins/krrxw | json-mask "url,object(content,attachments/url)"')
+  process.exit(error ? 1 : 0)
+}
+
+function pipeInput() {
+  return new Promise((resolve, reject) => {
+    process.stdin.resume()
+    process.stdin.setEncoding('utf8')
+
+    let data = ''
+    
+    process.stdin.on('data', chunk => {
+      data += chunk
+    })
+    
+    process.stdin.on('end', () => {
+      resolve(data)
+    })
+    
+    process.stdin.on('error', reject)
+  })
+}
+
+function readInputFile (inputFile, resolve, reject) {
+  const fs = require('fs').promises
```

Comment:
1. `require('fs').promises` is only available since Node v10. So this throws an error a few lines  below.
2. Let's put all `require()`s on top of the file. This will make it easier to transition to `import`s later.

Consider replacing this function altogether with the following on top of the file:

```js
const fs = require('fs')
const { promisify } = require('util')

const readFile = promisify(fs.readFile)
```

Btw, I'm drawing a blank on why `path.resolve` is necessary here. Can you remind me, please?

Reply:
I have run the tests on using `nve 8` and it ran just fine. What gives?!
I'll change it though.

Reply:
`path.resolve` resolves relative input paths to absolute ones. I am not really familiar with the node file functions, but isn't it better to hand in an absolute path to readFile just to be sure where the file is read from?

Reply:
I’ve tried both relative and absolute paths and readFile seems to work fine without a resolve, so I can’t seem to find a case where this is needed.